vllm/model_executor/layers/quantization/utils/w8a8_utils.py

```diff
@@ -138,8 +140,10 @@ def apply_fp8_linear(
             # A fix for discrepancy in scaled_mm which returns tuple
             # for torch < 2.5 and a single value in torch >= 2.5
             if type(output) is tuple and len(output) == 2:
-                return torch.narrow(output[0], 0, 0, input.shape[0])
-            return torch.narrow(output, 0, 0, input.shape[0])
+                output = output[0]
+            return (torch.narrow(
+                output, 0, 0, input.shape[0]) if not batched else output.view(
+                    input.shape[0], input.shape[1], weight.shape[1]))
```

Comment:
This assumes the input is 3D, while we do not check this. Maybe it would be better to do `output.view(*input.shape[:-1], weight.shape[1])`

Reply:
Fixed the check to verify that dim() == 3 since this seems to be the only other option in mllama MLP, and the only one I tested for